geoips/image_utils/maps.py

```diff
@@ -592,3 +599,9 @@ def draw_gridlines(mapobj, area_def, curr_ax, gridline_annotator, zorder=None):
     glnr.yformatter = LATITUDE_FORMATTER
     glnr.xlabel_style = {"rotation": 0}
     glnr.ylabel_style = {"rotation": 0}
+    # The following are the list of supported attributes which we can define in our
+    # gridline annotator. If specified, add these attributes to the x / y style dicts.
+    for style_arg in ["fontfamily", "fontstyle", "fontweight", "fontsize"]:
+        if style_arg in labels.keys():
+            glnr.xlabel_style[style_arg] = labels[style_arg]
+            glnr.ylabel_style[style_arg] = labels[style_arg]
```

Comment:
Why disallow all of the other options available for xlabel_style and ylabel_style? [See here](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.text.html).

Reply:
I was thinking we'd mainly just want to control what type of font modifications we would allow. However, after looking through those options you posted, I can see their use cases. I'll change the code to allow all of those options!